task_processing/task_processing/framework.py

```diff
@@ -1,225 +1,397 @@
-# -*- coding: utf-8 -*-
 from __future__ import absolute_import
 from __future__ import unicode_literals
 
+import copy
 import logging
+import random
+import threading
+import time
+import uuid
+from threading import Timer
 
 import mesos.interface
-import mesos.native
+import service_configuration_lib
 from mesos.interface import mesos_pb2
 
+from paasta_tools.frameworks.constraints import check_offer_constraints
```

Comment:
we'll get there - this is just a first pass. I'm not going to do everything in one PR